frontend/pages/index.js

```diff
@@ -41,27 +41,27 @@ const Home = () => {
    * @param ModalValue -The current value of the state
    */
   /* eslint-disable no-unused-vars*/
-  const triggerModal = (setModal, ModalValue) => {
+  function triggerModal(setModal, ModalValue) {
     /* eslint-disable no-unused-vars*/
-    setModal((ModalValue) => !ModalValue); //uses arrow function to make sure current value is used and not stale data
-  };
+    setModal(ModalValue => !ModalValue); //uses arrow function to make sure current value is used and not stale data
```

Comment:
The arrow function is not actually what makes sure the state is current, it's actually the way `setState` works.. Might want to re-examine this concept of `useState`

Reply:
So if I understand correctly by using the arrow function it grabs the current value immediately so it can then modify it, whereby if you dont do that you may not get the most up to date value as setState can execuate in its own time (I'm probably phrasing this a bit wierd).

Reply:
https://reactjs.org/docs/state-and-lifecycle.html